scripts/scripts.js

```diff
@@ -57,6 +58,51 @@ function setPageLanguage(param) {
   createMetadata('footer', '/footer');
 }
 
+export function pushToDataLayer(data) {
+  if (!data || !data.event) {
+    // eslint-disable-next-line no-console
+    console.error('The data layer event is missing');
+    return;
+  }
+
+  if (!window.adobeDataLayer) {
+    window.adobeDataLayer = [];
+    window.adobeDataLayerInPage = true;
+  }
+
+  window.adobeDataLayer.push(data);
+}
+
+export function trackProduct(product) {
+  // eslint-disable-next-line max-len
+  const isDuplicate = TRACKED_PRODUCTS.find((p) => p.platform_product_id === product.platform_product_id && p.variation_id === product.variation_id);
+  const isTrackedPage = getMetadata('analytics-tracking') === 'product';
```

Comment:
Maybe we can skip the rest if we find a duplicate.

Reply:
I realized we have duplicates in most cases.